LibSerialize.lua

```diff
@@ -1328,13 +1493,42 @@ function LibSerialize:SerializeEx(opts, ...)
     return ser._flushWriter()
 end
 
-function LibSerialize:Serialize(...)
-    return self:SerializeEx(defaultOptions, ...)
+function LibSerialize:SerializeEx(opts, ...)
+    local ser = CreateSerializer(opts)
+    return serializeOperation(ser, ...)
 end
 
-function LibSerialize:DeserializeValue(input)
-    local deser = CreateDeserializer(input)
+function LibSerialize:SerializeAsyncEx(opts, ...)
+    local ser = CreateSerializer(opts, true)
 
+    if opts.yieldOnElapsedTime and not opts.timeFn then
+        error("Async Mode operation with yieldOnElapsedTime requires timeFn option")
+    end
+
+    local thread = coroutine_create(serializeOperation)
+    local dots = {...}
+    -- return coroutine handler
+    return function()
+        local co_success, result = coroutine_resume(thread, ser, unpack(dots))
+        if not co_success then
+            error(result)
+        elseif coroutine_status(thread) ~= 'dead' then
+            return true
+        else
+            return false, result
+        end
+    end
+end
+
+function LibSerialize:SerializeAsync(...)
+    return self:SerializeAsyncEx(defaultAsyncOptions, ...)
```

Comment:
This is slightly misleading since you're technically going to get _all_ defaults, not just the async defaults. You could update this code to just pass `nil` and check `opts` against `nil` in the `createSerializer/createDeserializer` functions.

Reply:
Update the other variants of this function too.